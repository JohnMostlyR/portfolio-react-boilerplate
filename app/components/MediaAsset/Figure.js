import PropTypes from 'prop-types';
import styled from 'styled-components';
import { theme } from '../../styles/theme';

const { backgroundColor } = theme.project.media;

/* eslint-disable indent */
const Figure = styled.figure`
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 0;
  order: ${props => (props.isOdd ? 1 : 2)};
  position: relative;
  width: 100%;
  background-color: ${props =>
    props.isOdd ? backgroundColor.odd : backgroundColor.even};
  border-radius: 0.5rem;

  &::before {
    height: 0;
    left: 50%;
    position: absolute;
    top: 100%;
    transform: translateX(-50%);
    width: 0;
    content: '';
    border-bottom: 0;
    border-left: 0.75em solid transparent;
    border-top: 0.75em solid
      ${props => (props.isOdd ? backgroundColor.odd : backgroundColor.even)};
    border-right: 0.75em solid transparent;

    @media (min-width: 660px) {
      left: ${props => (props.isOdd ? '100%' : 'inherit')};
      right: ${props => (props.isOdd ? 'inherit' : '100%')};
      top: inherit;
      transform: translateX(0);
      border-bottom: 0.75em solid transparent;
      border-left: ${props =>
        props.isOdd ? `0.75em solid ${backgroundColor.odd}` : 0};
      border-top: 0.75em solid transparent;
      border-right: ${props =>
        props.isOdd ? 0 : `0.75em solid ${backgroundColor.even}`};
    }
  }

  @media (min-width: 660px) {
    width: auto;
  }
`;

Figure.propTypes = {
  isOdd: PropTypes.bool,
};

export default Figure;
