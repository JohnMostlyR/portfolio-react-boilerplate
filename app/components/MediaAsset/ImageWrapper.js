import PropTypes from 'prop-types';
import styled from 'styled-components';

const ImageWrapper = styled.div`
  height: ${({ imageDimensions: { height } = {} } = {}) =>
    height ? `${height}px` : 'auto'};
  width: ${({ imageDimensions: { width } = {} } = {}) =>
    width ? `${width}px` : '100%'};
  border-radius: 0.5rem;

  > img {
    border-radius: 0.5rem;
  }
`;

ImageWrapper.propTypes = {
  imageDimensions: PropTypes.shape({
    height: PropTypes.number,
    width: PropTypes.number,
  }),
};

export default ImageWrapper;
