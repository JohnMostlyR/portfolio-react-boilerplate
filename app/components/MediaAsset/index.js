import React from 'react';
import PropTypes from 'prop-types';

import Figure from './Figure';
import Body from './Body';
import ImageWrapper from './ImageWrapper';

function MediaAsset({ imageSource = {}, isOdd }) {
  const {
    file: {
      details: { image },
      url,
    },
    title = '',
  } = imageSource;
  return (
    <Figure isOdd={isOdd}>
      <Body>
        <ImageWrapper imageDimensions={image}>
          <img src={url} alt={title} />
        </ImageWrapper>
      </Body>
    </Figure>
  );
}

MediaAsset.propTypes = {
  imageSource: PropTypes.shape({
    file: PropTypes.shape({
      details: PropTypes.shape({
        image: PropTypes.shape({
          height: PropTypes.number,
          width: PropTypes.number,
        }),
      }),
      fileName: PropTypes.string,
      url: PropTypes.string,
    }),
    title: PropTypes.string,
  }).isRequired,
  isOdd: PropTypes.bool,
};

export default MediaAsset;
