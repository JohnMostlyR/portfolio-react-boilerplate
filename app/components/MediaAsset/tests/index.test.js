import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MediaAsset from '../index';

describe('MediaAsset', () => {
  const fixture = {
    file: {
      details: {
        image: {
          height: 10,
          width: 20,
        },
      },
      fileName: 'test-s.png',
      url: '/test-s.png',
    },
    title: 'Test image',
  };

  it('should render and match the snapshot', () => {
    const wrapper = shallow(<MediaAsset imageSource={fixture} isOdd />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should adopt the "idOdd" prop', () => {
    const isOdd = false;
    const wrapper = shallow(<MediaAsset imageSource={fixture} isOdd={isOdd} />);
    expect(wrapper.prop('isOdd')).toBe(isOdd);
  });
});
