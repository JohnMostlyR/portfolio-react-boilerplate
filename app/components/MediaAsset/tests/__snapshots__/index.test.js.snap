// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaAsset should render and match the snapshot 1`] = `
<Figure
  isOdd={true}
>
  <Body>
    <ImageWrapper
      imageDimensions={
        Object {
          "height": 10,
          "width": 20,
        }
      }
    >
      <img
        alt="Test image"
        src="/test-s.png"
      />
    </ImageWrapper>
  </Body>
</Figure>
`;
